### DISCUSSION TITLE
Admin token for demo.netbox.dev

### DISCUSSION THREAD

**Discussion by networkop** [netbox-community/netbox#8536] [Ideas]
Hi 👋 ,
I wanted to see if there was any appetite to enable an admin token on [demo.netbox.dev](https://demo.netbox.dev/)?
The use case would be to test API code against it. Similar to Cisco's devnet sandboxes or https://dev.megaport.com/
Another ulterior motive I've got is to include it in a book I'm writing about network automation. I'd like to be able to just give readers a URL + Token and let them test API calls against it (instead of explaining how to login and where to create a new token)

**Comment by markkuleinio** [comment]
Hi, why not teach them how to create the NetBox token via the API?

https://netbox.readthedocs.io/en/stable/release-notes/version-3.0/#rest-api-token-provisioning-5264

I see this is not properly documented in `/api/docs`.

**Reply by networkop** [reply]
ha, I didn't know about that! this could work

**Comment by jeremystretch** [comment]
The public demo instance is intended for incidental testing and evaluation, and is completely rebuilt with fresh data nightly. It does not include any default REST API tokens because any function which requires data to persist across builds should not rely on the demo instance.

**Reply by networkop** [reply]
yep, I get that. It's just that (for me) API tokens are an equivalent of admin credentials and you're provisioning them automatically, so I was wondering why not do the same for the token.

**Comment by DanSheps** [comment]
Alternatively, you could host your own instance of NetBox for your book.

Free branding/exposure for whatever website you have.

**Comment by networkop** [comment]
I think I got my answers. Thank you all for the discussion.
